internal/database/drivers/mongodb/mongodb.go

```diff
@@ -171,28 +171,31 @@ func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEvent
 	var numberOfDocuments int64
 	for _, collection := range collections {
 		var events []drivers.EiffelEvent
+		limit := request.PageSize - int32(len(allEvents))
```

Comment:
Side note: I don't think we're checking that request.PageSize is within a reasonable range. It's an int32 so a client could e.g. pass a negative number. Something to look into in a follow-up. Same thing with the `limit` and `levels` params for the POST /search endpoint where -1 has a specific meaning.

Reply:
We should check and return a proper error to the client. The API will return 0 events if it's negative in this case though due to the conditional on line 181